src/Services/Products/Distribt.Services.Products.BusinessLogic/DataAccess/ProductsReadStore.cs

```diff
@@ -27,10 +27,10 @@ public class ProductsReadStore : IProductsReadStore
     private const string CollectionName = "Products";
     private readonly IMongoDatabase _mongoDatabase;
 
-    public ProductsReadStore(IMongoDbConnectionProvider mongoDbConnectionProvider,
-        IOptions<DatabaseConfiguration> databaseConfiguration)
+    public ProductsReadStore(IMongoDbConnectionProvider mongoDbConnectionProvider, IOptions<DatabaseConfiguration> databaseConfiguration)
     {
-        _mongoClient = new MongoClient(mongoDbConnectionProvider.GetMongoUrl());
+        var mongoUrl = mongoDbConnectionProvider.GetMongoUrl().GetAwaiter().GetResult();
```

Comment:
this getmongo is used always with `.GetAwaiter().GetResult();` instead of do this all the time, i think is better to have the sync version even if it is in the interface `MongoUrl GetMongoUrlSync();` and do the `getawaiter.getresult` on that method

Or even use it outside of the constructor 🤔  but for that we will need a wrapper around the mongoclient, and something like 
```
public class DistribtMongoClient
{
 private readonly provider;
 
 public async Task<MongoClient> CreateClient(){
    return new MongoClient(await provider.getMongoURL())
 }
}
```


what do you think?